src/module/helpers/rolls.mjs

```diff
@@ -310,3 +310,201 @@ export class PowerRoll extends DSRoll {
+   * @param {string[]} [options.skills] - An array of skills that might be chosen
+   */
+  static async prompt(options = {}) {
+    const evaluation = options.evaluation ?? "message";
+    const formula = options.formula ?? "2d10";
+    if (!["none", "evaluate", "message"].includes(evaluation)) throw new Error("The `evaluation` parameter must be 'none', 'evaluate', or 'message'");
+    const flavor = options.flavor ?? game.i18n.localize("DRAW_STEEL.Roll.Project.Label");
+
+    const dialogContext = {
+      modChoices: Array.fromRange(3).reduce((obj, number) => {
+        obj[number] = number;
+        return obj;
+      }, {}),
+      bane: options.banes ?? 0,
+      edges: options.edges ?? 0
+    };
+
+    if (options.skills) {
+      dialogContext.skills = options.skills.reduce((obj, skill) => {
+        const label = ds.CONFIG.skills.list[skill]?.label;
+        if (!label) {
+          console.warn("Could not find skill" + skill);
+          return obj;
+        }
+        obj[skill] = label;
+        return obj;
+      }, {});
+    }
+
+    const content = await renderTemplate(systemPath("templates/rolls/prompt.hbs"), dialogContext);
+
+    const rollContext = await foundry.applications.api.DialogV2.prompt({
+      window: {title: "DRAW_STEEL.Roll.Project.Label"},
+      content,
+      ok: {
+        callback: (event, button, dialog) => {
+          const output = Array.from(button.form.elements).reduce((obj, input) => {
+            if (input.name) obj[input.name] = input.value;
+            return obj;
+          }, {});
+
+          return output;
+        }
+      }
+    });
+
+    const roll = new this(formula, options.data, {flavor, ...rollContext});
+
+    switch (evaluation) {
+      case "none":
+        return roll;
+      case "evaluate":
+        return roll.evaluate();
+      case "message":
+        return roll.toMessage();
+    }
+  }
+
+  /**
+   * Determines if this is a power roll with 2d10 base
+   */
+  get validProjectRoll() {
+    const firstTerm = this.terms[0];
+    return (firstTerm instanceof foundry.dice.terms.Die) && (firstTerm.faces === 10) && (firstTerm.number === 2);
+  }
+
+  /**
+   * Cancels out edges and banes to get the adjustment
+   * @returns {number} An integer from -2 to 2, inclusive
+   */
+  get netBoon() {
+    return this.options.edges - this.options.banes;
+  }
+
+  /**
+   * Total project progress accrued from this roll
+   */
+  get product() {
+    if (this._total === undefined) return undefined;
+    return Math.max(1, this.total);
+  }
+
+  /**
+   * Returns the natural result of the power roll
+   * @returns {number | undefined}
+   */
+  get naturalResult() {
+    return this.dice[0].total;
+  }
+
+  /**
+   * Determines if the natural result was a natural 20
+   * @returns {boolean | null} Null if not yet evaluated
+   */
+  get nat20() {
+    if ((this._total === undefined) || !this.validProjectRoll) return null;
+    return (this.dice[0].total >= 20);
+  }
```

Comment:
Is "nat 20" a term actually used in DS?

Regardless, could we adopt a convention for getters that ask a question like this - perhaps "isNat20" to make it seem less like a property and more like a consequence of other data on the roll.

Reply:
It is, although I need to double check if the current rules ever treat it differently from just being a critical (nat 19 or 20)